pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -228,6 +252,11 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
+	if instance.Status.ActivePlan.Name != planExecution.Name || instance.Status.ActivePlan.Namespace != planExecution.Namespace {
+		// this can happen for newly created PlanExecution where ActivePlan was not yet set to point to this instance
+		return reconcile.Result{}, fmt.Errorf("instance %s does not have ActivePlan pointing to PlanExecution %s, %s. Instead %s, %s. Retrying", instance.Name, planExecution.Name, planExecution.Namespace, instance.Status.ActivePlan.Name, instance.Status.ActivePlan.Namespace)
```

Comment:
You're already watching instance updates, so there is no need to re-trigger this reconciliation (it might create a loop)